gcp-client/pom.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ Copyright 2019 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ compliance with the License. You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under the License
+ is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing permissions and limitations under the
+ License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.google.graphite</groupId>
+    <artifactId>gcp-plugin-core-java</artifactId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+
+  <!--TODO(stephenashank) Prepare for maven release -->
+  <artifactId>gcp-client</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>GCP Plugin Clients</name>
+  <description>These classes provide shared GCP client functionality.</description>
+
+  <dependencies>
+    <dependency>
```

Comment:
I'd have to look more into a provided parent POM, but the parent POM for this repo could be modified to include a shared version number for the Google APIs. The kicker is that each client has different revisions for their versions, so it would look something like `<version>v1-rev57-${google.api.version}</version>`. I just tried and it's not possible to simply use the version number itself, the revision is required.

I think you bring up and interesting point, which is whether these APIs should depend on each other. Right now that isn't a concern but when the OAuth portion is added then that might be a concern here.

Reply:
I'm now using one central version defined in the parent pom of this client module, and added individual properties for the revisions.